torchvision/models/detection/ssd.py

```diff
@@ -0,0 +1,486 @@
+        cls_logits = nn.ModuleList()
+        for channels, anchors in zip(in_channels, num_anchors):
+            cls_logits.append(nn.Conv2d(channels, num_classes * anchors, kernel_size=3, padding=1))
+        _xavier_init(cls_logits)
+        super().__init__(cls_logits, num_classes)
+
+
+class SSDRegressionHead(SSDScoringHead):
+    def __init__(self, in_channels: List[int], num_anchors: List[int]):
+        bbox_reg = nn.ModuleList()
+        for channels, anchors in zip(in_channels, num_anchors):
+            bbox_reg.append(nn.Conv2d(channels, 4 * anchors, kernel_size=3, padding=1))
+        _xavier_init(bbox_reg)
+        super().__init__(bbox_reg, 4)
+
+
+class SSD(nn.Module):
+    __annotations__ = {
+        'box_coder': det_utils.BoxCoder,
+        'proposal_matcher': det_utils.Matcher,
+    }
+
+    def __init__(self, backbone: nn.Module, anchor_generator: DBoxGenerator, size: Tuple[int, int], num_classes: int,
+                 image_mean: Optional[List[float]] = None, image_std: Optional[List[float]] = None,
+                 score_thresh: float = 0.01,
+                 nms_thresh: float = 0.45,
+                 detections_per_img: int = 200,
+                 iou_thresh: float = 0.5,
+                 topk_candidates: int = 400,
+                 positive_fraction: float = 0.25):
+        super().__init__()
+
+        # Use dummy data to retrieve the feature map sizes to avoid hard-coding their values
+        device = next(backbone.parameters()).device
+        tmp_img = torch.zeros((1, 3, size[1], size[0]), device=device)
+        tmp_sizes = [x.size() for x in backbone(tmp_img).values()]
+        out_channels = [x[1] for x in tmp_sizes]
+
+        assert len(out_channels) == len(anchor_generator.aspect_ratios)
+
+        self.backbone = backbone
+
+        self.anchor_generator = anchor_generator
+
+        self.box_coder = det_utils.BoxCoder(weights=(10., 10., 5., 5.))
+
+        # Estimate num of anchors based on aspect ratios: 2 default boxes + 2 * ratios of feaure map.
+        self.num_anchors = [2 + 2 * len(r) for r in anchor_generator.aspect_ratios]
+        self.head = SSDHead(out_channels, self.num_anchors, num_classes)
+
+        self.proposal_matcher = det_utils.DBoxMatcher(iou_thresh)
+
+        if image_mean is None:
+            image_mean = [0.485, 0.456, 0.406]
+        if image_std is None:
+            image_std = [0.229, 0.224, 0.225]
+        self.transform = GeneralizedRCNNTransform(min(size), max(size), image_mean, image_std,
+                                                  size_divisible=1, fixed_size=size)
+
+        self.score_thresh = score_thresh
+        self.nms_thresh = nms_thresh
+        self.detections_per_img = detections_per_img
+        self.topk_candidates = topk_candidates
+        self.neg_to_pos_ratio = (1.0 - positive_fraction) / positive_fraction
+
+        # used only on torchscript mode
+        self._has_warned = False
```

Comment:
FYI: This was originally used because torchscript had a bug and would warn at every forward. I think this bug has been fixed since then so this is probably not needed anymore

Reply:
Sounds good, will remove.